tests/unit_tests/components/config.cpp

```diff
@@ -0,0 +1,143 @@
+#include <iostream>
+
+#include <unistd.h> // unlink, rmdir
+
+#include "common.hpp"
+#include "errors.hpp"
+#include "components/config.hpp"
+#include "components/logger.hpp"
+#include "utils/file.hpp"
+
+#include "common/test.hpp"
+
+using namespace polybar;
+
+namespace {
+
+  class temp_directory {
+  public:
+    temp_directory(string directory) {
+      if (file_util::is_absolute(directory)) {
+        m_directory = directory;
+      } else {
+        m_directory = "/tmp/" + directory;
+      }
+
+      if (m_directory.length() < 6 || m_directory.compare(m_directory.length() - 6, 6, "XXXXXX") != 0) {
```

Comment:
We can also always directly add `_XXXXXX` to the path, why would any caller of this function want to manually add the Xs.

Reply:
Hypothetically, they might want more than six "X"s or a dot instead of an underscore. But even if I did assume it was always added by the function, I'd still want an assert in there, just to be safe

Reply:
I don't think we need to be that flexible in our tests, the main point here is to have a unique temporary directory. Whether or not it has an underscore or a period doesn't matter, I think.